src/book/02-system/01-nubots/06-localisation.mdx

```diff
@@ -2,6 +2,26 @@
+
+The localisation method used to estimate world to field space uses a Particle Filter. The Particle Filter estimates the transform using particles. Initially, the Particle Filter creates and normally distributes a number of particles, each representing a guess of the world to field transform. Each particle is equally likely to be correct. The particles are then updated when we see goal posts. The difference between the observed and expected goal post measurements for each particle is calculated. A number of random "rogue" particles are added to avoid local maxima. Based on the difference calculated each particle is awarded a weight and ranked with the worst particles discarded.
+
+# Implementation
+
+The [Robot Particle Localisation](https://github.com/NUbots/NUbots/tree/master/module/localisation/RobotParticleLocalisation) module uses the [Particle Filter](https://github.com/NUbots/NUbots/blob/master/shared/utility/math/filter/eigen/ParticleFilter.h) utility with template type [RobotModel](https://github.com/NUbots/NUbots/blob/master/module/localisation/RobotParticleLocalisation/src/RobotModel.h). The module emits a [Field](https://github.com/NUbots/NUbots/blob/master/shared/message/localisation/Field.proto) message that can then be used in the rest of the system. This should be used in conjunction with [SensorFilter](https://github.com/NUbots/NUbots/tree/master/module/platform/darwin/SensorFilter), which handles robot odometry and gives the world to robot matrix.
+
+The module will set parameters using the given [YAML](https://github.com/NUbots/NUbots/blob/master/module/localisation/RobotParticleLocalisation/data/config/RobotParticleLocalisation.yaml) file. At this time, the filter is initialised with particles based on the parameters of the YAML file.
+
+RobotParticleLocalisation runs a time update at a specified frequency and a measurement update. Since the world to field transform does not change over time, i.e. the world and the field are stationary, the time update does nothing. The measurement update runs every time a [Goals](https://github.com/NUbots/NUbots/blob/master/shared/message/vision/Goal.proto) message is received from [vision](/system/nubots/vision), alongside a [FieldDescription](https://github.com/NUbots/NUbots/blob/master/shared/message/support/FieldDescription.proto) message that describes what the field the robot is playing on looks like. In this update, the weights of the particles are adjusted based on the goals message.
```

Comment:
It's great that we're saying how the localisation modules deal with incoming messages here, but it makes me curious as to when and why the localisation module outputs messages.